src/Components/CameraFeed/videoPlayer.tsx

```diff
@@ -0,0 +1,198 @@
+  playerRef: React.RefObject<HTMLVideoElement>;
+  streamUrl: string;
+  className?: string;
+  onPlay?: () => void;
+  onEnded?: () => void;
+  onWaiting?: () => void;
+  onSuccess?: (resp: any) => void;
+  onError?: (err: any) => void;
+}
+
+export default function VideoPlayer(props: VideoPlayerProps) {
+  const wsRef = useRef<WebSocket>();
+  const playerRef = props.playerRef;
+  let mediaSource: MediaSource;
+  let mseSourceBuffer: SourceBuffer;
+  let buf: Uint8Array;
+  let bufLen = 0;
+
+  const pushPacket = () => {
+    if (mseSourceBuffer.updating) return;
+
+    try {
+      if (bufLen > 0) {
+        // If there's data in the buffer to append
+        const data = buf.slice(0, bufLen);
+        bufLen = 0; // Reset buffer length
+        mseSourceBuffer.appendBuffer(data); // Append data to SourceBuffer
+      } else if (mseSourceBuffer.buffered && mseSourceBuffer.buffered.length) {
+        // If no new data to append, check if there's buffered data in SourceBuffer
+        const end =
+          mseSourceBuffer.buffered.end(mseSourceBuffer.buffered.length - 1) -
+          15;
+        const start = mseSourceBuffer.buffered.start(0);
+        if (end > start) {
+          // Remove older data from the SourceBuffer to free up space
+          mseSourceBuffer.remove(start, end);
+          mediaSource.setLiveSeekableRange(end, end + 15);
+        }
+      }
+    } catch (e) {
+      console.debug(e);
+      props.onError?.(e);
+    }
+  };
+
+  const readPacket = (event: MessageEvent<any>) => {
+    if (mseSourceBuffer.updating || bufLen > 0) {
+      // Buffer data if SourceBuffer is updating or buffer has data
+      const b = new Uint8Array(event.data);
+      buf.set(b, bufLen);
+      bufLen += b.byteLength;
+    } else {
+      try {
+        // Append data directly if SourceBuffer is ready
+        mseSourceBuffer.appendBuffer(event.data);
+      } catch (e) {
+        console.debug(e);
+        props.onError?.(e);
+      }
+    }
+  };
+
+  const cleanup = () => {
+    console.debug("Cleaning up video player");
+    if (wsRef.current) {
+      wsRef.current.close();
+    }
+    if (mseSourceBuffer) {
+      mseSourceBuffer.abort();
+    }
+    if (playerRef.current) {
+      playerRef.current.pause();
+      playerRef.current.src = "";
+      playerRef.current.srcObject = null;
+    }
+  };
+
+  const starthls = () => {
+    console.debug("Broken os/browser, falling back to hls");
+    try {
+      if (!playerRef.current || !props.streamUrl) return;
+      const url = new URL(props.streamUrl);
+      if (url.protocol === "wss:") {
+        url.protocol = "https:";
+      }
+      url.pathname = url.pathname.replace("mse", "hls/live/index.m3u8");
+      playerRef.current.src = url.toString();
+      playerRef.current.onplaying = () => {
+        props.onSuccess?.(undefined);
+      };
+    } catch (err) {
+      console.debug(err);
+      props.onError?.(err);
+    }
+  };
+
+  const startmse = () => {
```

Comment:
https://docs.aws.amazon.com/prescriptive-guidance/latest/best-practices-cdk-typescript-iac/typescript-best-practices.html#naming-conventions

```suggestion
  const startMSE = () => {
```